nexus/src/app/instance.rs

```diff
@@ -1488,8 +1484,12 @@ impl super::Nexus {
                         Some(Ok(WebSocketMessage::Frame(_) | WebSocketMessage::Ping(_) | WebSocketMessage::Pong(_))) => {}
                     }
                 }
-                result = nexus_write => {
-                    result?;
+                result = nexus_reserve => {
+                    let permit = result?;
+                    let message = buffered_output
+                        .take()
+                        .expect("nexus_reserve is only active when buffered_output is Some");
+                    permit.send(message)?.await?;
```

Comment:
Note a change in behavior here -- the flush involved here now blocks the whole task. I *think* this is OK since we're on the local network. We could do this differently but if we decide to do that, I think we should do a bit of an overall -- spawn 2-3 different tasks and communicate across them using channels.

Reply:
Yeah, it should be acceptable to let the task wait here; I imagine the performance impact being negligible (unless you've observed otherwise). If we end up wanting to further optimize this endpoint's throughput/latency in the future, we can certainly revisit it.